src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -182,90 +188,47 @@ protected virtual void ProcessTypes(AssemblyDefinition assembly, XPathNavigator
                         continue;
                 }
 
-                TypeDefinition type = assembly.MainModule.GetType(fullname);
-
-                if (type == null && assembly.MainModule.HasExportedTypes)
-                {
-                    foreach (var exported in assembly.MainModule.ExportedTypes)
-                    {
-                        if (fullname == exported.FullName)
-                        {
-                            var resolvedExternal = ProcessExportedType(exported, assembly, typeNav);
-                            if (resolvedExternal != null)
-                            {
-                                type = resolvedExternal;
-                                break;
-                            }
-                        }
-                    }
-                }
+                TypeDesc type = CustomAttributeTypeNameParser.GetTypeByCustomAttributeTypeName(assembly, fullname, throwIfNotFound: false);
```

Comment:
This has different semantics around nested types at minimum; I used it as a shortcut, but if we want to be bug-for-bug compatible with illink, we'll need to build another emulator for whatever Cecil accepts there (it's another spot where the documentation of the illink file format is "whatever Cecil accepts" without any specification of what that is).

Could you add a TODO?